### PR TITLE
user roles

### PR DESCRIPTION
**Pull request by jeremie-j**
Original Comment moved #110 

## In this pr
Added roles, with a mapping of services (biography, masterclasses...) and right (viewer, editor)
ex: Admin `{"USER_MANAGMENT": "EDITOR"}`

Added a table to store theses roles depending of the academy, and permit to be flexible as the service right mapping can be customized.

Added a property to users, to refer to a role id

Added a check in custom security, check that the user has the necessary role to access a service

### CODE REVIEW

File: backend/scripts/create_academy.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        "service_rights": {
+            Service.MASTERCLASS: Right.VIEWER,
+            Service.BIOGRAPHY: Right.EDITOR,
+        },
+    },
+    "Film crew member": {
+        "name": "Film crew member",
+        "description": "Can be assigned to a masterclass to upload videos",
+        "service_rights": {
+            Service.MASTERCLASS: Right.VIEWER,
```
Comment (domino659) [inline backend/scripts/create_academy.py:34]: Il lui manque pas un RIGHT pour l'upload ?

File: backend/src/entities/authentification/dependencies.py [hunk 2]
```diff
@@ -32,25 +33,27 @@ def __call__(
...
 
-    @staticmethod
-    def is_admin(user: User):
-        """
-        Check if the user is an admin.
-
-        Args:
-            user (User, optional): The user to check.
+    def _has_right(self, role_id: int) -> bool:
+        if not self.service_rights:
```
Comment (domino659) [inline backend/src/entities/authentification/dependencies.py:45]: manque commentaire, mais pas forcément essentiel

File: backend/scripts/create_academy.py (hunk 1 above)
Comment (domino659) [reply backend/scripts/create_academy.py:34]: Ou c'est custom a chaque academie ?

File: backend/src/entities/authentification/dependencies.py (hunk 2 above)
Comment (jeremie-j) [reply backend/src/entities/authentification/dependencies.py:45]: docstring ?

Comment (domino659) [reply backend/src/entities/authentification/dependencies.py:45]: Oui

**Review by domino659 — APPROVED** [review]

**Pull request merged**